File: backend/domain/paymentcategory/entity.go
package paymentcategory

import (
	"time"

	"github.com/google/uuid"
)

type PaymentCategory struct {
	PaymentCategoryId   uuid.UUID
	UserId              string
	PaymentCategoryName string
	IsNeeds             bool
	ColorName           ColorName
	ColorTone           ColorTone
	CreatedAt           time.Time
	UpdatedAt           time.Time
}

func NewPaymentCategory(userId, paymentCategoryName string, isNeeds bool, colorName ColorName, colorTone ColorTone) PaymentCategory {
	paymentCategoryId := uuid.New()
	createdAt := time.Now()
	updatedAt := time.Now()

	return PaymentCategory{
		PaymentCategoryId:   paymentCategoryId,
		UserId:              userId,
		PaymentCategoryName: paymentCategoryName,
		IsNeeds:             isNeeds,
		ColorName:           colorName,
		ColorTone:           colorTone,
		CreatedAt:           createdAt,
		UpdatedAt:           updatedAt,
	}
}

File: frontend/src/domain/payment-category/entity.ts
import { z } from "zod";
import {
	type ColorName,
	type ColorTone,
	colorNameSchema,
	colorToneSchema,
} from "./value-object";

export type PaymentCategory = {
	paymentCategoryId: string;
	userId: string;
	paymentCategoryName: string;
	isNeeds: boolean;
	colorName: ColorName;
	colorTone: ColorTone;
	createdAt: Date;
	updatedAt: Date;
};

export const paymentCategorySchema: z.ZodType<PaymentCategory> = z.object({
	paymentCategoryId: z.string().uuid(),
	userId: z.string(),
	paymentCategoryName: z.string(),
	isNeeds: z.boolean(),
	colorName: colorNameSchema,
	colorTone: colorToneSchema,
	createdAt: z.date(),
	updatedAt: z.date(),
});
